p2p/example/libp2p/autodiscovery/main.go

```diff
@@ -3,64 +3,52 @@ package main
 import (
 	"context"
 	"fmt"
-	"math/rand"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/ElrondNetwork/elrond-go/config"
 	"github.com/ElrondNetwork/elrond-go/display"
 	"github.com/ElrondNetwork/elrond-go/p2p"
 	"github.com/ElrondNetwork/elrond-go/p2p/libp2p"
-	"github.com/ElrondNetwork/elrond-go/p2p/libp2p/discovery"
-	"github.com/ElrondNetwork/elrond-go/p2p/loadBalancer"
-	libp2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
 )
 
-var r *rand.Rand
+func createMockNetworkArgs() libp2p.ArgsNetworkMessenger {
+	return libp2p.ArgsNetworkMessenger{
+		Context:       context.Background(),
+		ListenAddress: libp2p.ListenLocalhostAddrWithIp4AndTcp,
+		P2pConfig: config.P2PConfig{
+			Node: config.NodeConfig{},
+			KadDhtPeerDiscovery: config.KadDhtPeerDiscoveryConfig{
+				Enabled:                          true,
+				RefreshIntervalInSec:             2,
+				RandezVous:                       "",
+				InitialPeerList:                  nil,
+				BucketSize:                       100,
+				RoutingTableRefreshIntervalInSec: 10,
+			},
+			Sharding: config.ShardingConfig{
+				Type: p2p.DisabledSharder,
+			},
+		},
+	}
+}
 
 //The purpose of this example program is to show what happens if a peer connects to a network of 100 peers
 func main() {
-	r = rand.New(rand.NewSource(time.Now().UnixNano()))
 	startingPort := 32000
 
-	randezVous := "test"
-	arg := discovery.ArgKadDht{
-		PeersRefreshInterval: time.Second,
-		RandezVous:           randezVous,
-		InitialPeersList:     nil,
-		BucketSize:           100,
-		RoutingTableRefresh:  time.Minute,
-	}
-	peerDiscovery, _ := discovery.NewKadDhtPeerDiscoverer(arg)
-	advertiser, _ := libp2p.NewNetworkMessenger(
-		context.Background(),
-		startingPort,
-		genPrivKey(),
-		nil,
-		loadBalancer.NewOutgoingChannelLoadBalancer(),
-		peerDiscovery,
-		libp2p.ListenLocalhostAddrWithIp4AndTcp,
-		0,
-	)
+	advertiser, err := libp2p.NewNetworkMessenger(createMockNetworkArgs())
+	fmt.Println(err)
```

Comment:
Do we need this print?

Reply:
good catch, no. was used during debugging